src/components/WeatherWidget/WeatherWidget.test.tsx

```diff
@@ -199,6 +214,29 @@ describe('WeatherWidget', () => {
     expect(screen.getByTestId('weatherWidget_input')).toHaveValue('90210')
   })
 
+  test('user can retry fetching the weather forecast', async () => {
+    const user = userEvent.setup()
+    const mockConsoleError = jest.fn()
+    jest.spyOn(console, 'error').mockImplementation(mockConsoleError)
+
+    renderWithMySpaceAndModalContext(
+      <WeatherWidget widget={mockWeatherWidgetWithIncorrectZipcode} />
+    )
+
+    // Click the retry button
+    await waitFor(() => {
+      expect(
+        screen.getByLabelText('Retry fetching weather')
+      ).toBeInTheDocument()
+    })
+    await user.click(screen.getByLabelText('Retry fetching weather'))
+
+    // Since the mock widget that is being used has an invalid url, we expect the request to fail. Checking
+    // that there is an error logged to the console is important because it means that, when the Retry
+    // button was clicked, the request was made in the useWeather hook and failed.
+    expect(mockConsoleError).toHaveBeenCalled()
```

Comment:
Thought(non-blocking): I don't think it's required but I do have a preference for checking for specific error messages vs just that `console.error` was called since technically if another call to `console.error` happens this test would still pass.

Reply:
Yeah that's a great point. Just made an update!